### PR TITLE
[WIP] Update C_SHARP.md

### PR DESCRIPTION
**Pull request by joaoburatto**
## Summary

Updates the programming guidelines.

### CODE REVIEW

File: Documents/C_SHARP.md [hunk 1]
```diff
@@ -78,23 +78,23 @@ For example:
...
 public class MyClass 
 {
-    public int publicField;
-    private int packagePrivate;
-    private int myPrivate;
-    protected int myProtected;
+    public int PublicField;
+    private int _packagePrivate;
+    private int _myPrivate;
+    protected int _myProtected;
```
Comment (Ryan089) [inline Documents/C_SHARP.md:84]: None of these adhere to the standard described in line 74. It needs to be amended as well, to reflect that public fields will be PascalCase, and private fields will be camelCase with a leading underscore (if that is what the scheme is intended to be).

File: Documents/C_SHARP.md [hunk 2]
```diff
@@ -433,7 +433,7 @@ Avoid throwing exceptions. Instead log and error. Methods returning values shoul
 ```csharp
 public List<Transform> FindAThing(int arg){
     ...
-    if (notFound) {
+    if (_notFound) {
```
Comment (Ryan089) [inline Documents/C_SHARP.md:436]: Does not comply with Brace Style policy.

File: Documents/C_SHARP.md [hunk 3]
```diff
@@ -443,7 +443,7 @@ public List<Transform> FindAThing(int arg){
...
```
Comment (Ryan089) [inline Documents/C_SHARP.md:446]: Does not comply with Brace Style policy.

File: Documents/C_SHARP.md [hunk 4]
```diff
@@ -496,10 +496,10 @@ Prefer RequireComponent and GetComponent over AddComponent. Having the component
 ```csharp
```
Comment (Ryan089) [inline Documents/C_SHARP.md:496]: Line 493: There should be no apostrophe in "lets"

File: Documents/C_SHARP.md [hunk 5]
```diff
@@ -496,10 +496,10 @@ Prefer RequireComponent and GetComponent over AddComponent. Having the component
...
 public class : MonoBehaviour
 {
-    private AudioSource audioSource;
+    private AudioSource _audioSource;
 
     private void Start() {
```
Comment (Ryan089) [inline Documents/C_SHARP.md:501]: Does not comply with Brace Style policy.

File: Documents/C_SHARP.md [hunk 6]
```diff
@@ -509,10 +509,10 @@ public class : MonoBehaviour
 [RequireComponent(typeof(AudioSource))]
 public class : MonoBehaviour
 {
-    private AudioSource audioSource;
+    private AudioSource _audioSource;
 
     private void Start() {
```
Comment (Ryan089) [inline Documents/C_SHARP.md:514]: Does not comply with Brace Style policy.

File: Documents/C_SHARP.md [hunk 7]
```diff
@@ -525,11 +525,11 @@ Sometimes it's just nice to see them for debugging, even if we don't change them
...
 ```csharp
-public GameObject someMember;
+public GameObject SomeMember;
 ```
 
 **BETTER:**
 ```csharp
-[SerializeField] private GameObject someMember;
+[SerializeField] private GameObject _someMember;
 public GameObject SomeMember => someMember;
```
Comment (Ryan089) [inline Documents/C_SHARP.md:534]: Need to add underscore to someMember.

**Review by Ryan089 — COMMENTED** [review]
Minor inconsistencies identified for consideration.

**Comment by joaoburatto** [comment]
Will be back into this soon.

**Pull request closed without merging**